src/test/java/uk/gov/digital/ho/proving/income/audit/AuditArchiveServiceIT.java

```diff
@@ -0,0 +1,203 @@
+            .andExpect(jsonPath("$.eventIds.*", containsInAnyOrder("corr-id-1")))
+            .andExpect(jsonPath("$.result", is("PASS")))
+            .andExpect(jsonPath("$.resultDate", is("2019-01-01")))
+            .andRespond(withSuccess());
+        mockAuditService
+            .expect(requestTo(containsString("/archive")))
+            .andExpect(method(POST))
+            .andExpect(jsonPath("$.nino", is("nino_2")))
+            .andExpect(jsonPath("$.lastArchiveDate", is(endOfArchive)))
+            .andExpect(jsonPath("$.eventIds.*", containsInAnyOrder("corr-id-2")))
+            .andExpect(jsonPath("$.result", is("PASS")))
+            .andExpect(jsonPath("$.resultDate", is("2019-01-01")))
+            .andRespond(withSuccess());
+
+        auditArchiveService.archiveAudit();
+    }
+
+    @Test
+    public void archiveAudit_noResponse_errorReturned() {
+        String errorRequest = fileUtils.buildRequest("corr-id-1", "2019-01-02 11:00:00.000", "nino_1");
+        String auditHistory = String.format("[%s]", errorRequest);
+        mockAuditService
+            .expect(requestTo(containsString("/history")))
+            .andExpect(method(POST))
+            .andRespond(withSuccess(auditHistory, APPLICATION_JSON));
+
+        mockAuditService
+            .expect(requestTo(containsString("/archive")))
+            .andExpect(method(POST))
+            .andExpect(jsonPath("$.nino", is("nino_1")))
+            .andExpect(jsonPath("$.eventIds.*", containsInAnyOrder("corr-id-1")))
+            .andExpect(jsonPath("$.result", is("ERROR")))
+            .andExpect(jsonPath("$.resultDate", is("2019-01-02")))
+            .andRespond(withSuccess());
+
+        auditArchiveService.archiveAudit();
+    }
+
+    @Test
+    public void archiveAudit_errorThenNotFound_bestResultReturned() {
+        String errorRequest = fileUtils.buildRequest("corr-id-1", "2019-01-02 11:00:00.000", "nino_1");
+        String notFoundRequest = fileUtils.buildRequest("corr-id-2", "2019-01-03 12:00:00.000", "nino_1");
+        String notFoundResponse = fileUtils.buildResponseNotFound("corr-id-2", "2019-01-03 13:00:00.000");
+        String auditHistory = String.format("[%s, %s, %s]", errorRequest, notFoundRequest, notFoundResponse);
+        mockAuditService
+            .expect(requestTo(containsString("/history")))
+            .andExpect(method(POST))
+            .andRespond(withSuccess(auditHistory, APPLICATION_JSON));
+
+        mockAuditService
+            .expect(requestTo(containsString("/archive")))
+            .andExpect(method(POST))
+            .andExpect(jsonPath("$.nino", is("nino_1")))
+            .andExpect(jsonPath("$.eventIds.*", containsInAnyOrder("corr-id-1", "corr-id-2")))
+            .andExpect(jsonPath("$.result", is("NOTFOUND")))
+            .andRespond(withSuccess());
+
+        auditArchiveService.archiveAudit();
+    }
+
+    @Test
+    public void archiveAudit_multipleMatchingResults_latestResultReturned() {
```

Comment:
Hopefully you can point it out to me but I cannot see how this test asserts that the latest result is returned.